core/src/Kokkos_View.hpp

```diff
@@ -1741,10 +1741,13 @@ class View : public ViewTraits<DataType, Properties...> {
   // Conversion to MDSpan
   template <class OtherElementType, class OtherExtents, class OtherLayoutPolicy,
             class OtherAccessor,
-            typename = std::enable_if_t<std::is_assignable_v<
-                mdspan<OtherElementType, OtherExtents, OtherLayoutPolicy,
-                       OtherAccessor>,
-                typename Impl::MDSpanViewTraits<traits>::mdspan_type>>>
+            class Dummy = typename Impl::MDSpanViewTraits<traits>::mdspan_type,
+            typename    = std::enable_if_t<std::conditional_t<
+                std::is_same_v<Impl::UnsupportedKokkosArrayLayout, Dummy>,
+                std::false_type,
+                std::is_assignable<mdspan<OtherElementType, OtherExtents,
+                                          OtherLayoutPolicy, OtherAccessor>,
+                                   Dummy>>::value>>
```

Comment:
```suggestion
            class ImplNaturalMDSpanType = typename Impl::MDSpanViewTraits<traits>::mdspan_type,
            typename    = std::enable_if_t<std::conditional_t<
                std::is_same_v<Impl::UnsupportedKokkosArrayLayout, ImplNaturalMDSpanType>,
                std::false_type,
                std::is_assignable<mdspan<OtherElementType, OtherExtents,
                                          OtherLayoutPolicy, OtherAccessor>,
                                   ImplNaturalMDSpanType>>::value>>
```

This might be a bit clearer what it's doing since it's more specific